baseplate/core.py

```diff
@@ -123,6 +124,145 @@ def from_upstream(cls, trace_id, parent_id, span_id, sampled, flags):
         return cls(trace_id, parent_id, span_id, sampled, flags)
 
 
+class AuthenticationContextFactory(object):
+    """Building factory for :py:class:`AuthenticationContext` objects, handles
+    context dependency for the secrets store.
+
+    This factory should be passed into the constructor of any upstream
+    :doc:`integrations <integration/index>` so that it is aware of the
+    application's secret store and can create the authentication context for
+    the incoming requests.
+
+    :param baseplate.secrets.SecretsStore secrets_store: the application's
+        defined secrets store, where application secret lookups should be made
+    """
+    def __init__(self, secrets_store=None):
+        self.secrets = secrets_store
+
+    def make_context(self, token):
+        """:py:class:`AuthenticationContext` builder
+
+        :param token: JWT value originating from the Authentication service
+            either directly or from an upstream service
+        :rtype: :py:class:`AuthenticationContext`
+        """
+        return AuthenticationContext(token=token, secrets=self.secrets)
+
+
+class AuthenticationContext(object):
+    """Wrapper for the contextual authentication information
+
+    :param str token: the JWT value for the currently propagated authentication
+        context
+    :param baseplate.secrets.SecretsStore secrets_store: the application's
+        defined secrets store, where application secret lookups should be made
+    """
+    def __init__(self, token=None, secrets=None):
+        self.token = token
+        self.secrets = secrets
```

Comment:
Probably, I left it accessible because it doesn't feel too scary to expose and I can't see much of an argument about what exposing it could hurt, I may spend some time and rework this and try to refactor how the location of the public key is defined (rather than having it hard-coded as it is now).